crates/iceberg/src/error.rs

```diff
@@ -44,6 +44,10 @@ pub enum ErrorKind {
     ///
     /// This error is returned when given iceberg feature is not supported.
     FeatureUnsupported,
+    /// Arrow compute error.
+    ///
+    /// Error returned by compute function in arrow.
+    ArrowCompute,
```

Comment:
`ArrowCompute` is not an error code that can take actions. How about remove it or replace it by an actionable error code.